Tests/Virtual Time/VirtualTimers.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  VirtualTimers.swift
+//  StreamChatClientTests
+//
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+@testable import StreamChatClient
+
+struct VirtualTimeTimer: StreamChatClient.Timer {
+    
+    static var time: VirtualTime!
+    
+    static func schedule(timeInterval: DispatchTimeInterval, queue: DispatchQueue, onFire: @escaping () -> Void) {
+        _ = Self.time.scheduleTimer(interval: TimeInterval(dispatchTimeInterval: timeInterval),
+                                    repeating: false,
+                                    callback: { _ in onFire() })
+    }
+    
+    static func scheduleRepeating(timeInterval: DispatchTimeInterval,
+                                  queue: DispatchQueue,
+                                  onFire: @escaping () -> Void) -> TimerControl {
+        Self.time.scheduleTimer(interval: TimeInterval(dispatchTimeInterval: timeInterval),
+                                repeating: true,
+                                callback: { _ in onFire() })
+    }
+}
+
+extension VirtualTime.Timer: TimerControl {
+    func resume() {
+        isActive = true
+    }
+    
+    func suspend() {
+        isActive = false
+    }
+}
+
```

Comment:
I think there's nothing wrong about setting `isActive` directly in the tests. No need to complicate things. This is not public api.

Reply:
Why would we use internal API in tests? It's safer to be close to real-life usage scenario, in which we'll never access `isActive` directly. I also disagree it's complicating, on the contrary I think accessing `isActive` is more complicated than just `timer.suspend()` or `resume()`

Reply:
Bahadir, we're talking about `VirtualTimer` which is used solely in tests, never in production. If mutating a variable directly bothers you for some reason, I'll add the functions you want 🤷‍♂️

Reply:
That's a fair point, but then why do we have the extensions? If we have functions to mutate these variables, we should only use them.
Even if it's testing something only used for testing, API could be consistent.
I accept your point though, it's up to you, we can keep it.

Reply:
We have it because both simple and repeating timers are represented by the same object in tests, while in the production, we handle them differently. `DispatchQueue.asyncAfter` for simple timers, and `DispatchSourceTimer` for repeating timers.

I'll change it so it has the api you want.